parse/lex_internal_test.go

```diff
@@ -243,3 +243,119 @@ func TestLex_defaultStateFn(t *testing.T) {
 		defaultStateFn(s, e)
 	}
 }
+
+func TestNumberStateFn(t *testing.T) {
+	tests := []struct {
+		input        string
+		expectedType TokenType
+		expectedVal  string
+	}{
+		{"1", Int, "1"},
+		{"0", Int, "0"},
+		{"123", Int, "123"},
+		{"9990909", Int, "9990909"},
+		{"+909", Int, "+909"},
+		{"-909", Int, "-909"},
+	}
```

Comment:
Later we should test when illegal token comes to `numberStateFn`

Reply:
I agree!

Reply:
thx.  i fix my bug and add testcase